### PR TITLE
Support @strawberry.type

### PR DESCRIPTION
**Pull request by koxudaxi**
support @strawberry.type as dataclass

## How to build it
https://koxudaxi.github.io/pydantic-pycharm-plugin/development/

### CODE REVIEW

File: scripts/build_changelog.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import re
+import xml.etree.ElementTree
+from enum import Enum
+from html.parser import HTMLParser
+from pathlib import Path
+from typing import List
+
```
Comment (patrick91) [inline scripts/build_changelog.py]: what's this file for? 😊

Comment (koxudaxi) [reply scripts/build_changelog.py]: Oh, I forgot to remove it.
The scripts generate changelog pages from histores.
https://koxudaxi.github.io/pydantic-pycharm-plugin/changelog/

File: testData/mock/stub/builtins.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+"""
+The Mock is for testing pydantic with builtins
```
Comment (patrick91) [inline testData/mock/stub/builtins.py]: 👀

Comment (koxudaxi) [reply testData/mock/stub/builtins.py]: I have changed the name to `strawberry`.
We will need the unittest. I keep this file.

File: resources/META-INF/plugin.xml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<idea-plugin url="https://github.com/strawberry-graphql/strawberry-pycharm-plugin" require-restart="true">
+    <id>rocks.strawberry</id>
+    <name>Strawberry-Graphql</name>
```
Comment (patrick91) [inline resources/META-INF/plugin.xml]:
```suggestion
    <name>Strawberry-GraphQL</name>
```

where is this name used? :)

File: resources/META-INF/plugin.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    <version>0.0.1</version>
+    <vendor email="koaxudai@gmail.com">TODO Koudai Aono @koxudaxi</vendor>
```
Comment (patrick91) [inline resources/META-INF/plugin.xml:5]: will this email be used for support? we can create one @strawberry.rocks if needed :)

**Review by patrick91 — APPROVED** [review]
Awesome stuff! This looks good to me :)

Is there anything you need from us to get this published, shall we merge it and deal with the rest later?

File: resources/META-INF/plugin.xml (hunk 3 above)
Comment (koxudaxi) [reply resources/META-INF/plugin.xml]: Here!!
https://plugins.jetbrains.com/plugin/12861-pydantic

File: resources/META-INF/plugin.xml (hunk 4 above)
Comment (koxudaxi) [reply resources/META-INF/plugin.xml:5]: great!!
This plugin.xml will be a detailed page in the marketplace.
https://plugins.jetbrains.com/plugin/12861-pydantic

**Comment by koxudaxi** [comment]
@patrick91 
Thank you for reviewing PR.
> Is there anything you need from us to get this published, shall we merge it and deal with the rest later?

Jetbrains Account, plugin Id, and a token to publish the plugin.
https://plugins.jetbrains.com/docs/marketplace/plugin-upload.html
https://account.jetbrains.com/login

I will add this CI file after I have dinner. 😉 
https://github.com/koxudaxi/poetry-pycharm-plugin/blob/da2d98bbf3853711770cb32b1d7d4f39b844dcd1/.github/workflows/publish.yml#L74-L78


-----

The pydantic plugin is my first plugin project.
I used a plugin.xml to configure the plugin.
But, we should move some settings to `build.gradle`.
 [gradle-intellij-plugin](https://github.com/JetBrains/gradle-intellij-plugin) can descript few configs in `build.gradle` that  instead of`plugin.xml`
gradle-intellij-plugin  provides dynamically configuration. (we can write gralde script or kotlin DSL)
For example, 
Poetry plugin use `patchPluginXml` in build.gradle. It's my second project.
This is not a dynamic attribute. But, I feel the style is better than writing it in `plugin.xml`
https://github.com/koxudaxi/poetry-pycharm-plugin/blob/da2d98bbf3853711770cb32b1d7d4f39b844dcd1/build.gradle#L28-L31

This idea is not required for first release.
We should add it TODO list 📝

File: resources/META-INF/plugin.xml (hunk 4 above)
Comment (patrick91) [reply resources/META-INF/plugin.xml:5]: got it, we can update the email later :)

**Comment by patrick91** [comment]
I've added a token in the configuration: 
image

We can use GitHub issues for TODOs :) I'll add you to this repo, feel free to merge this PR when you want.

**Pull request merged**